hydroflow/examples/kvs_bench/buffer_pool/mod.rs

```diff
@@ -0,0 +1,99 @@
+mod serialization;
+
+pub use serialization::AutoReturnBufferDeserializer;
+
+use std::{
+    cell::{Ref, RefCell, RefMut},
+    rc::{Rc, Weak},
+};
+
+pub type BufferType = [u8; 1024];
+
+#[derive(Debug)]
+pub struct BufferPool {
+    buffers: Vec<Rc<RefCell<BufferType>>>,
+}
+
+#[derive(Clone, Debug)]
+pub struct AutoReturnBufferInner {
+    pub collector: Weak<RefCell<BufferPool>>,
+    pub inner: Rc<RefCell<BufferType>>,
+}
```

Comment:
Would like a bit more comments on how this is set up. lot of nesting rcrefcells

Reply:
I added some comments, let me know what you think